src/workflows/registration.ts

```diff
@@ -113,3 +116,19 @@ export async function getSignableRegistrationOnchain(
     payload_hash: response.data.payload_hash,
   };
 }
+
+async function isUserRegistered(
+  userAddress: string,
+  usersApi: UsersApi,
+): Promise<boolean> {
+  try {
+    await usersApi.getUsers({ user: userAddress });
+    return true;
+  } catch (error) {
+    const axiosError = error as AxiosError;
+    if (axiosError.response?.status === 404) {
+      return false;
+    }
+    throw error;
```

Comment:
Do we want to throw an error or return false?
I am asking because I wonder if we should keep trying to proceed with the **registerUser** if `isUserRegistered` throw an error.

Reply:
As `isUserRegistered` isn't essential for the registration to work, we can ignore other errors and let the workflow continue.

Reply:
ok. So if we throw the error, the workflow would stop performing as well, have to try-catch in `registerOffchainWorkflow` then?

Reply:
We're ignoring errors from this api now ([45a061b](https://github.com/immutable/imx-core-sdk/pull/85/commits/45a061b114645dae7008ac7b54ab7f7fb4ff37da)) if they aren't 404.